mesh/submesh/psubmesh.cpp

```diff
@@ -167,6 +167,7 @@ ParSubMesh::ParSubMesh(const ParMesh &parent, SubMesh::From from,
 
    // Add boundaries
    {
+      int num_of_faces_or_edges = (Dim == 2) ? NumOfEdges : NumOfFaces;
```

Comment:
You could also use `Mesh::GetNumFaces()` for this if you want to

Reply:
I think this is what I planned to do in the original implementation (with the bug). But internally it feels wrong to use `Mesh::GetNumFaces()` because of the 2D/3D discrepancy of the term "Face".